components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -47,7 +47,18 @@ open class GleanInternalAPI internal constructor () {
     private val gleanLifecycleObserver by lazy { GleanLifecycleObserver() }
 
     // `internal` so this can be modified for testing
-    internal var initialized = false
+    internal enum class GleanInitStates {
+        // This state represents glean as totally uninitialized
+        UNINITIALIZED,
+        // This state represents glean initializing, where we can send pings but not record metrics
```

Comment:
nit: "... initializing , where pings can be triggered by internal code but external callers of the specific metrics API are not allowed to record".